### DISCUSSION TITLE
Higher-level init function

### DISCUSSION THREAD

**Discussion by KyleAMathews** [electric-sql/electric#1020] [Feature requests]
There's more concerns accumulating for initing & maintaining a connection to Electric from the client e.g. picking a db name, and creating the client and now as of 0.9.4, connecting and re-connecting on the token expiring. This is what my initElectric function looks like — https://github.com/KyleAMathews/electric-query/commit/8721875d14c40797018e668886f9b1c68b6bf39d

It seems like these concerns could be pulled into electric itself as most everyone would do the same thing?

**Comment by thruflo** [comment]
Would be nice to have convenience functions for this (and for handling the auth reconnect).

**Comment by kevin-dp** [comment]
Hi @KyleAMathews,
As you raised before on Discord, JWTs are usually short-lived so using the JWT's expiration to dictate the end of the session is not a good idea. Based on your input, we decided to use the JWT only for authentication and we will need to introduce support for sessions. So you should not worry about the JWT expiring as we no longer close the connection when the JWT expires (the JWT's expiration is merely used as a deadline for authenticating). When we add support for sessions, programmers will need to handle expiration and renewal and we will document that accordingly.

Does this make sense?

Regards,
Kevin

**Reply by KyleAMathews** [reply]
ahh ok — good to know. So I just need to listen then for the user returning to the app (`visibilitychange`) and then checking then if I'm disconnected and just doing the same `.connect(token)` call again?

**Reply by kevin-dp** [reply]
Yes, ofcourse if you re-connect the token you pass needs to still be valid so it may require fetching a new token.

**Reply by KyleAMathews** [reply]
ok, this is what I changed it to — look good? https://github.com/KyleAMathews/electric-query/commit/02f98841df8dda65b323aea02154d3a2f3736b7a

**Reply by KyleAMathews** [reply]
actually switched to https://github.com/GoogleChromeLabs/page-lifecycle from visibilitychange as that event doesn't capture when you alt-tab (or otherwise) enter an app, only when switching between tabs.

**Reply by KyleAMathews** [reply]
It's working well now with the page-lifecycle lib. I open my browser to a electric app and it now immediately reconnects.

**Reply by kevin-dp** [reply]
Looks good!

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
